library/src/main/java/com/wultra/android/powerauth/networking/utils/AppUtils.kt

```diff
@@ -34,4 +36,75 @@ class AppUtils {
             return appContext.packageManager.getPackageInfo(appContext.packageName, flags)
         }
     }
-}
\ No newline at end of file
+}
+
+class ConnectionMonitor(context: Context) {
```

Comment:
This thing is a bit buggy.
**Bug 1:**
1. User agent is used by subclasses of `Api` class. Every created instance fills its `userAgent` val during its `init` (it's either passed from outside or a default is created).
2. Once instance of `Api` is created, the value is filled, it never changes.
3. Obtaining connectivity this way is pointless, the value is relevant only to the moment in time when `UserAgent` instance is created. All `Api` instance requests will have the same value forever.

**Bug 2:**
1. Every call to `UserAgent.libraryDefault(Context)` creates a new instance of `ConnectionMonitor`.
2. On non-ancient version of Android it registers `ConnectivityManager.NetworkCallback` in its `init`.
4. Inside system's `ConnectivityManager` references to these callbacks are strongly held. They are never released on its own and continue to be called until either the app exits or `unregisterNetworkCallback()` is called.
5. In worst case scenario (unless instance of `UserAgent` is shared by consumer code) for every instance of `Api` subclass one `NetworkCallback` and transitively one `ConnectionMonitor` is leaked.
6. If we also consider Bug 1, registering `NetworkCallback` is completely pointless here.

Put together:
- `ConnectionMonitor` can't leak. If there's register there has to be unregister at some point. This might require bigger changes further up the call chain.
- If a current value of the connectivity is desired then the whole handling (including the usage) of `UserAgent` has to be changed. Either use a stored value that actually changes when connectivity changes or query the connectivity state when userAgent is requested (this way you can completely omit registering a `NetworkCallback`).